dll/win32/mpr/wnet.c

```diff
@@ -2800,10 +2806,11 @@ DWORD WINAPI WNetDisconnectDialog1A( LPDISCDLGSTRUCTA lpConnDlgStruct )
  */
 DWORD WINAPI WNetDisconnectDialog1W( LPDISCDLGSTRUCTW lpConnDlgStruct )
 {
-    FIXME( "(%p): stub\n", lpConnDlgStruct );
-
-    SetLastError(WN_NO_NETWORK);
-    return WN_NO_NETWORK;
+#ifdef __REACTOS__
+    HRESULT WINAPI SHDisconnectNetDrives(PVOID Unused);
+	
+    return SHDisconnectNetDrives(NULL);
```

Comment:
What should be done here instead is a LoadLibrary of the dll that exports this API, followed by a GetProcAddress to get the function pointer. At the end, release the loaded dll with FreeLibrary.

Reply:
Before I make another commit, does this look like what you wanted?
#ifdef __REACTOS__
    HMODULE hDll = LoadLibraryW(L"netplwiz.dll");
    static BOOL (WINAPI *pSHDisconnectNetDrives)(PVOID);
    pSHDisconnectNetDrives = GetProcAddress(hDll, "SHDisconnectNetDrives");
    
    return pSHDisconnectNetDrives(NULL);
 #endif

Reply:
`#ifdef __REACTOS__`

Reply:
@JaredSmudde : almost: a more correct code would be (note that the FreeLibrary is before the 'return' instruction):
```c
#ifdef __REACTOS__
    DWORD dwRet;
    HMODULE hDll = LoadLibraryW(L"netplwiz.dll");
    static BOOL (WINAPI *pSHDisconnectNetDrives)(PVOID);
    pSHDisconnectNetDrives = GetProcAddress(hDll, "SHDisconnectNetDrives");
    
    dwRet = pSHDisconnectNetDrives(NULL);
    
    FreeLibrary(hDll);
    return dwRet;
 #endif
```
This may be further improved by loading only once the dll and caching the imported procedure, etc...

Reply:
Is delay load an option here?